src/history/hs.c

```diff
@@ -1046,36 +1049,47 @@ __hs_delete_key_int(WT_SESSION_IMPL *session, uint32_t btree_id, const WT_ITEM *
     int cmp, exact;
 
     hs_cursor = session->hs_cursor;
-    hs_cursor->set_key(hs_cursor, btree_id, key, WT_TS_NONE, 0);
-    /*
-     * FIXME: This should be done in a loop in order to deal with the possibility of racing with
-     * other inserts. Perhaps further generalize the cursor positioning function.
-     */
+    WT_RET(__wt_scr_alloc(session, 0, &srch_key));
+
+    hs_cursor->set_key(hs_cursor, btree_id, key, WT_TS_NONE, (uint64_t)0);
```

Comment:
Why are you casting to (uint64_t)?

Reply:
I had to do it here: https://github.com/wiredtiger/wiredtiger/pull/5362#discussion_r388557382
So it should also be necessary here.

Reply:
Also we're doing it in master for lookaside: https://github.com/wiredtiger/wiredtiger/blob/master/src/cache/cache_las.c#L831